### PR TITLE
Check for no_proxy env to disable proxy in libwebsockets

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

**Comment by tribal-tec** [comment]
retest this please

**Comment by tribal-tec** [comment]
Updated

**Comment by tribal-tec** [comment]
retest this please

**Comment by tribal-tec** [comment]
retest this please

**Comment by tribal-tec** [comment]
Jenkins error, I created a ticket. Please approve nonetheless :)

File: rockets/clientContext.cpp [hunk 1]
```diff
@@ -73,19 +133,33 @@ lws* ClientContext::startHttpRequest(const http::Method method,
...
 }
 
 std::unique_ptr<ws::Connection> ClientContext::connect(
     const std::string& uri, const std::string& protocol)
 {
+#if LWS_LIBRARY_VERSION_NUMBER < 2004000
+    const bool recreateVhost = wsProtocolName != protocol;
+#endif
     wsProtocolName = protocol;
     protocols[0].name = wsProtocolName.c_str();
```
Comment (rdumusc) [inline rockets/clientContext.cpp]: please also move this hacky reassignement of protocols[0].name into the #if LWS_LIBRARY_VERSION_NUMBER < 2004000, I just re-tested that it is not needed with newer versions which probably correctly use connectInfo.protocol instead

Comment (tribal-tec) [reply rockets/clientContext.cpp]: Done

**Comment by rdumusc** [comment]
Perfect thanks!

**Pull request merged**